bundles/core/org.eclipse.smarthome.core/src/main/java/org/eclipse/smarthome/core/net/NetUtil.java

```diff
@@ -92,65 +87,26 @@ public String getPrimaryIpv4HostAddress() {
      */
     @Deprecated
     public static String getLocalIpv4HostAddress() {
-        try {
-            String hostAddress = null;
-            final Enumeration<NetworkInterface> interfaces = NetworkInterface.getNetworkInterfaces();
-            while (interfaces.hasMoreElements()) {
-                final NetworkInterface current = interfaces.nextElement();
-                if (!current.isUp() || current.isLoopback() || current.isVirtual()) {
-                    continue;
-                }
-                final Enumeration<InetAddress> addresses = current.getInetAddresses();
-                while (addresses.hasMoreElements()) {
-                    final InetAddress current_addr = addresses.nextElement();
-                    if (current_addr.isLoopbackAddress() || (current_addr instanceof Inet6Address)) {
-                        continue;
-                    }
-                    if (hostAddress != null) {
-                        LOGGER.warn("Found multiple local interfaces - ignoring {}", current_addr.getHostAddress());
-                    } else {
-                        hostAddress = current_addr.getHostAddress();
-                    }
-                }
-            }
-            return hostAddress;
-        } catch (SocketException ex) {
-            LOGGER.error("Could not retrieve network interface: {}", ex.getMessage(), ex);
-            return null;
-        }
+        return new NetUtil().getPrimaryIpv4HostAddress();
     }
 
-    private String getFirstLocalIPv4Address() {
-        try {
-            String hostAddress = null;
-            final Enumeration<NetworkInterface> interfaces = NetworkInterface.getNetworkInterfaces();
-            while (interfaces.hasMoreElements()) {
-                final NetworkInterface current = interfaces.nextElement();
-                if (!current.isUp() || current.isLoopback() || current.isVirtual()) {
-                    continue;
-                }
-                final Enumeration<InetAddress> addresses = current.getInetAddresses();
-                while (addresses.hasMoreElements()) {
-                    final InetAddress current_addr = addresses.nextElement();
-                    if (current_addr.isLoopbackAddress() || (current_addr instanceof Inet6Address)) {
-                        continue;
-                    }
-                    if (hostAddress != null) {
-                        LOGGER.warn("Found multiple local interfaces - ignoring {}", current_addr.getHostAddress());
-                    } else {
-                        hostAddress = current_addr.getHostAddress();
-                    }
-                }
-            }
-            return hostAddress;
-        } catch (SocketException ex) {
-            LOGGER.error("Could not retrieve network interface: {}", ex.getMessage(), ex);
+    private CidrAddress getAnyAssignedIPAddress(boolean preferIPv4) {
```

Comment:
You are only calling this method with `true` which is hardcoded in the sources. Should we offer this parameter as an option for the user to choose?

Reply:
The API can be used for both yes. And in the future we will probably switch the default to IPv6. But this is the default address that we pick here, that is used if the user didn't configure anything. Why should this be configurable?

Reply:
Because the user might prefer to use IPv6 now? And if you only want to provide this option in the future, then please remove the parameter for now.

Reply:
If the user wants ipv6 then he just chooses from the list that we present in paperui. Personally I would just pick the first IP we get from java to bind to. I just tried to be backwards compatible here by selecting an ipv4. Please be aware that no matter if we pick ipv4 or ipv6 as default, both protocols will work anyway.

Reply:
If that is the case then I do not see the need for this parameter. I it's only about backward compatibility that you prefer IPv4 here then you can remove this parameter and prefer IPv4 nevertheless. In my opinion a method with a boolean parameter that is only called with `true` is superfluous.

Reply:
The idea is to remove the parameter and to not prefer any IP version. I do not see why ipv4 should be preferred here, I know some pretty good ipv6 only setup. And the IP version doesn't actually matter, it is the interface that we bind to. At least on Linux such a socket works for ipv4+ipv6.

Reply:
Fine for me. But please have a look at my netmask comment below.